src/load-styles.js

```diff
@@ -0,0 +1,62 @@
+import { CDN_HOST, getBody, getHeaders, getVersion, ONE_YEAR } from './common.js';
+
+const CSS = 'application/javascript';
+
+export async function handleLoadStylesRequest (httpApi) {
+
+  const versionsList = await httpApi.fetch(`versions-list.json`, CDN_HOST);
+  if (versionsList == null) {
+    return httpApi.sendResponse({ body: '', status: 500 });
+  }
+
+  const version = getVersion(versionsList, httpApi.getQueryParam('version'));
+
+  const depsManifest = await httpApi.fetch(`deps-manifest-${version.resolved}.json`, CDN_HOST);
+  if (depsManifest == null) {
+    return httpApi.sendResponse({ body: '', status: 500 });
+  }
+
+  const response = await getResponse(version, depsManifest, httpApi);
+
+  return httpApi.sendResponse(response);
+}
+
+export async function getResponse (version, depsManifest, httpApi) {
+
+  if (version.resolved == null) {
+    return {
+      status: 400,
+      body: getBody([
+        `/* Unknown version */`,
+      ]),
+      headers: getHeaders({ type: CSS, maxAge: 0 }),
+    };
+  }
+
+  const versionComment = `/* VERSION: ${version.resolved} */`;
+
+  const headers = getHeaders({
+    type: CSS,
+    maxAge: (version.requested === version.resolved) ? ONE_YEAR : 0,
+  });
+
+  let stylesBlocks;
+  if (depsManifest.manifestVersion !== '2') {
```

Comment:
Maybe this should lead to a 404 error: the user is asking for a resource that doesn't exist for the specified version

Reply:
I wondered about that. I guess we could change this.

Reply:
Done